packages/next-app/src/components/FilterDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+"use client";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { ChevronDownIcon, EraserIcon } from "@radix-ui/react-icons";
+import { useState } from "react";
+
+import { toSlug, useAdminTools } from "#/contexts/AdminToolsContext";
+
+export interface IFilter {
+  name: string;
+  options: string[];
+}
+
+interface IFilterDropdown {
+  filters: IFilter[];
+}
+
+export function FilterDropdown({ filters }: IFilterDropdown) {
+  const { changeSelectedFilters, clearSelectedFilter } = useAdminTools();
+  return (
+    <DropdownMenu.Root>
+      <DropdownMenu.Trigger asChild>
+        <button
+          className="flex items-center gap-2 rounded-[4px] bg-white px-2 text-sm leading-none text-gray-400 outline-none"
+          aria-label="Customise options"
+        >
+          <span>Filter by</span>
+          <ChevronDownIcon />
+        </button>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Portal>
+        <DropdownMenu.Content
+          sideOffset={5}
+          className="rounded-md bg-white px-5 py-2 will-change-[opacity,transform] data-[side=top]:animate-slideDownAndFade data-[side=right]:animate-slideLeftAndFade data-[side=bottom]:animate-slideUpAndFade data-[side=left]:animate-slideRightAndFade"
+        >
+          {filters.map((filter) => {
+            const [filterValue, setFilterValue] = useState("");
```

Comment:
This should've triggered an eslint error, but this breaks one of the rules of hooks and could lead to bugs, see [this](https://legacy.reactjs.org/docs/hooks-rules.html)